src/eureka/S3_data_reduction/s3_reduce.py

```diff
@@ -402,29 +406,38 @@ def reduce(eventlabel, ecf_path=None, s2_meta=None, input_meta=None):
                                 and meta.oneoverf_corr is not None):
                             # save current flux into an array for
                             # plotting 1/f correction comparison
                             flux_w_oneoverf = np.copy(data.flux.values[i])
 
                         # Determine centroid position
-                        # We do this twice. First a coarse estimation,
-                        # then a more precise one.
-                        # Use the center of the frame as an initial guess
-                        centroid_guess = [data.flux.shape[1]//2,
-                                          data.flux.shape[2]//2]
-                        # Do a 2D gaussian fit to the whole frame
-                        position, extra = \
-                            centerdriver.centerdriver('fgc',
-                                                      data.flux.values[i],
-                                                      centroid_guess, 0, 0, 0,
-                                                      mask=None, uncd=None,
-                                                      fitbg=1, maskstar=True,
-                                                      expand=1.0, psf=None,
-                                                      psfctr=None, i=i, m=m,
-                                                      meta=meta)
+                        if (not hasattr(meta, 'ctr_guess') or
+                                meta.ctr_guess is None):
+                            # If meta.ctr_guess is None, first do a coarse
+                            # centroiding using the center of the frame as an
+                            # initial guess
+                            centroid_guess = [data.flux.shape[1]//2,
+                                              data.flux.shape[2]//2]
+                            # Do a 2D gaussian fit to the whole frame
+                            position, extra = \
+                                centerdriver.centerdriver('fgc',
+                                                          data.flux.values[i],
+                                                          centroid_guess,
+                                                          0, 0, 0,
+                                                          mask=None, uncd=None,
+                                                          fitbg=1,
+                                                          maskstar=True,
+                                                          expand=1.0, psf=None,
+                                                          psfctr=None,
+                                                          i=i, m=m,
+                                                          meta=meta)
+                        else:
+                            guess = np.array(meta.ctr_guess)[::-1]
+                            trim = np.array([meta.ywindow[0], meta.xwindow[0]])
+                            position = guess - trim
```

Comment:
I'm not keen on requiring a fixed `position` when a guess is provided.  If the user supplies a guess centroid, they should also have the option to feed the guess into `centerdriver`.

Reply:
Well the guess just replaces the first round of centroiding which was done on the entire frame. For MIRI, centroiding on the entire frame is a really bad idea because the frame is so large and there can easily be multiple sources, and even if there is only one source in the field the centroid gets strongly affected by the very non-uniform background.

To be clear, a second round of centroiding is still done in lines 451-464 in the exact same was as was done before (using a smaller subarray around the guess), just using the manually guessed value instead of the full-frame automatic centroiding. And if no guess is provided, then the code runs the exact same as before.

Reply:
Ah, ok.  Thanks for clarifying.  Now it makes sense.